src/test/run-make/rustc-macro-dep-files/Makefile

```diff
@@ -3,6 +3,10 @@
 # FIXME(eddyb) provide `HOST_RUSTC` and `TARGET_RUSTC`
 # instead of hardcoding them everywhere they're needed.
 all:
-	$(BARE_RUSTC) foo.rs --out-dir $(TMPDIR)
+    ifeq ($(IS_MUSL_HOST),1)
+	    $(BARE_RUSTC) $(RUSTFLAGS) -Clinker=$(RUSTC_LINKER) foo.rs --out-dir $(TMPDIR)
```

Comment:
Hm how come this change to the test was necessary?

Reply:
`$(BARE_RUSTC)` doesn't set correct linker like `$(RUST)` does exactly [here](https://github.com/rust-lang/rust/blob/c1d2d83ca3b5155468ab96b09a7c54568449b137/src/test/run-make-fulldeps/tools.mk#L14).
It just picks `cc` from the path which happens to be Ubuntu toolchain which targets glibc and messes everything.
It's fine when making static binaries because Rust [will override](https://github.com/rust-lang/rust/blob/master/src/librustc_target/spec/linux_musl_base.rs#L9-L26) it's default libs and thus avoid glibc. The problem is one cannot build `dylibs` or `proc_macro` crates when targetting static musl so I'm using dynamic target.

Reply:
Ok thanks for the explanation!

I think though that this test should probably just be disabled in situations like this? Most of these tests are only really engineered to run in "vanilla" situations which correspond to default setups, so in this case I think it'd be best to just disable this test in the case that we're doing some sort of cross compilation or cross-host compilation (aka host != build)

Reply:
Good idea.
I'll introduce environment variable (with more generic name) in `compiletest` and guard this test. Obliviously with a comment why this won't work when `host` and `build` are different.